### PR TITLE
feat: dockerfile

### PR DESCRIPTION
**Pull request by y-eight**
## Motivation

Prepared Dockerfile for building the `sparrow` image.

## Changes

For additional information look at the commits.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,12 @@
+FROM cgr.dev/chainguard/go:latest as gobuild
```
Comment (lvlcn-t) [inline Dockerfile:1]: As @puffitos mentioned in https://github.com/caas-team/sparrow/pull/22#discussion_r1408883502

> Maybe use the go1.21 image equivalent, no need for latest :)

We should define the go version here

Comment (y-eight) [reply Dockerfile]: since chainguards images by tag are not available for free I have switched to golang image for the app build.

Comment (y-eight) [reply Dockerfile]: golang version will be passed to the dockerfile by --build-arg

**Comment by NiklasTreml** [comment]
Scratch images come without certificates, please add them to the runner image, otherwise we won't be able to use https:
```dockerfile
FROM alpine:latest as certs
RUN apk add --no-cache ca-certificates
...
# In your runner image
COPY --from=certs /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
```

**Comment by y-eight** [comment]
Ive added the ca certs directly from gobuild stage

**Review by puffitos — APPROVED** [review]
I dislike having a variable go version (we don't have to abstract/ leave everything to the pipeline), but other than that, lgtm.

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+    --shell "/sbin/nologin" \
+    --no-create-home \
+    --uid 65532 \
+    sparrow
+
+WORKDIR /app
+ADD . .
+RUN go mod download
+RUN go mod verify
+RUN CGO_ENABLED=0 go build -ldflags "-s -w -X main.version=${VERSION}" -o sparrow .
```
Comment (NiklasTreml) [inline Dockerfile:16]: Please also add the `static` flag to the linker

```bash
CGO_ENABLED=0 go build -a -ldflags '-s -w -extldflags "-static" -X main.version=0.0.1' -o app ./main.go
```

**Review by NiklasTreml — CHANGES_REQUESTED** [review]

**Pull request closed without merging**